storybookjs/storybook
# Controls: Fix select / multiselect when value contains multiple spaces

Closes #17051.



## Analysis

Select and multiselect controls won’t work when selecting an option with multiple spaces, causing the select to reset and the URL reference to be removed or added as "undefined", in the case of multiselect. This is due to the options not receiving the value prop. According to the HTML specs:

> The value of an [option](https://html.spec.whatwg.org/multipage/form-elements.html#the-option-element) element is the [value](https://html.spec.whatwg.org/multipage/form-elements.html#attr-option-value) of the value content attribute, if there is one, or, if there is not, the value of the element's [text](https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-text) IDL attribute.

So HTML is involved, which is notoriously unreliable about multiple white spaces.

## What I did

I've added the value to both the select and multiselect components.

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`.
2. Open Storybook in your browser.
3. Head to addons > controls > basics > undefined story or `/?path=/story/addons-controls-basics--undefined`.
4. Scroll down to the select control, click on it and select the option "double space".
5. Do the same for the multiselect control.
6. It should be selected.

Or just run the E2E tests I've added.



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
